### PR TITLE
4.0.0 Changelog

### PR DESCRIPTION
**Pull request by thatmidwesterncoder**
Adding the changelog entry for version 4.0.0

### CODE REVIEW

**Review by HarrisonWAffel — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,35 @@
+## v4.0.0 (February 2, 2024)
+
+FEATURES:
+
+* Added support for `machine_selector_files` on . See https://github.com/rancher/terraform-provider-rancher2/pull/1225
+* Added support for `graceful_shutdown_timeout` for vSphere nodes. See https://github.com/rancher/terraform-provider-rancher2/pull/1228
+* Added `inherited_cluster_roles` attribute to global_role. See https://github.com/rancher/terraform-provider-rancher2/pull/1242
+* Added `project_id` syntax for `rancher2_project_role_template_binding`. See https://github.com/rancher/terraform-provider-rancher2/pull/1272
```
Comment (jiaqiluo) [inline CHANGELOG.md]: It looks like the PR adds an example project_id to the description, so I suggest moving it to the enhancement section.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,35 @@
...
+* Set `protect-kernel-defaults` on rancher_v2 clusters. See https://github.com/rancher/terraform-provider-rancher2/pull/1244
+* Updated `default_system_registry` behavior at `rancher2_app_v2` resource. See https://github.com/rancher/terraform-provider-rancher2/pull/1265
+* Updated RKE config machine pool schema and structure. See https://github.com/rancher/terraform-provider-rancher2/pull/1253
+* [Docs] updated the documentation for `rancher_cluster_v2`. See https://github.com/rancher/terraform-provider-rancher2/pull/1300
+* [Docs] Remove `keypair_name` from `amazonec2_config`. See https://github.com/rancher/terraform-provider-rancher2/pull/942
+
+
+BUGFIXES:
+
+* Fixed msc old schema to have correct type. See https://github.com/rancher/terraform-provider-rancher2/pull/1223
```
Comment (jiaqiluo) [inline CHANGELOG.md]:
```suggestion
* Fixed `machine_selector_config` old schema to have the correct type. See https://github.com/rancher/terraform-provider-rancher2/pull/1223
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,3 +1,35 @@
...
+* Allow machine pools to be created and scaled down to 0. See https://github.com/rancher/terraform-provider-rancher2/pull/1232
+* Fixed the issue where the for_each in rancher2_registry keeps changing the order of the list that has not changed. See https://github.com/rancher/terraform-provider-rancher2/pull/1268
```
Comment (jiaqiluo) [inline CHANGELOG.md]:
```suggestion
* Fixed the issue where the for_each in `rancher2_registry` keeps changing the order of the list that has not changed. See https://github.com/rancher/terraform-provider-rancher2/pull/1268
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,3 +1,35 @@
...
+* Fixed double base64-encoding of `ca_bundle` field. See https://github.com/rancher/terraform-provider-rancher2/pull/1296
+* [Docs] Fixed description for using `kms_key` `eks_config_v2`. See https://github.com/rancher/terraform-provider-rancher2/pull/892
+* [Docs] Revert "Remove OpenStack keypair_name references". See https://github.com/rancher/terraform-provider-rancher2/pull/1235
```
Comment (jiaqiluo) [inline CHANGELOG.md]: how about moving it to the enhancement section? 
```suggestion
* Add the `keypair_name` argument  to the `openstack_config` machine config and node template.  See https://github.com/rancher/terraform-provider-rancher2/pull/1235
```

**Review by jiaqiluo — CHANGES_REQUESTED** [review]

**Review by jiaqiluo — APPROVED** [review]

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,3 +1,35 @@
+## v4.0.0 (February 2, 2024)
```
Comment (jiaqiluo) [inline CHANGELOG.md]: The date needs to be confirmed and updated if needed after the release.

**Review by jiaqiluo — CHANGES_REQUESTED** [review]
"request changes" here to avoid merging by accident. :)

**Review by jiaqiluo — APPROVED** [review]

**Pull request merged**